server/data/data-api/src/main/java/org/apache/james/probe/DataProbe.java

```diff
@@ -49,6 +52,15 @@ public FluentDataProbe addDomain(String domain) throws Exception {
             dataProbe.addDomain(domain);
             return this;
         }
+
+        public FluentDataProbe addDomainIfNotExists(String domain) throws Exception {
```

Comment:
I would rather make this the default behaviour. WDYT?

Reply:
Agree.